tests/test_measurements_collapse.py

```diff
@@ -68,55 +61,63 @@ def test_measurement_collapse_bitflip_noise(backend):
         output = c.add(gates.M(0, 1, p0=0.2, collapse=True))
 
 
-@pytest.mark.parametrize("density_matrix", [False, True])
+@pytest.mark.parametrize("density_matrix", [True, False])
 @pytest.mark.parametrize("effect", [False, True])
 def test_measurement_result_parameters(backend, effect, density_matrix):
     c = models.Circuit(4, density_matrix=density_matrix)
     if effect:
         c.add(gates.X(0))
     r = c.add(gates.M(0, collapse=True))
     c.add(gates.RX(1, theta=np.pi * r.symbols[0] / 4))
+    if not density_matrix:
+        c.add(gates.M(0))
 
     target_c = models.Circuit(4, density_matrix=density_matrix)
     if effect:
         target_c.add(gates.X(0))
         target_c.add(gates.RX(1, theta=np.pi / 4))
+    if not density_matrix:
+        target_c.add(gates.M(0))
 
-    final_state = backend.execute_circuit(c, nshots=1)[0]
+    final_state = backend.execute_circuit(c, nshots=1)
     target_state = backend.execute_circuit(target_c)
+    if not density_matrix:
+        final_state = final_state.samples()[0]
+        target_state = target_state.samples()[0]
     backend.assert_allclose(final_state, target_state)
 
 
 def test_measurement_result_parameters_random(backend):
-    initial_state = random_statevector(2**4, backend=backend)
+    initial_state = random_density_matrix(2**4, backend=backend)
     backend.set_seed(123)
-    c = models.Circuit(4)
+    c = models.Circuit(4, density_matrix=True)
     r = c.add(gates.M(1, collapse=True))
     c.add(gates.RY(0, theta=np.pi * r.symbols[0] / 5))
     c.add(gates.RX(2, theta=np.pi * r.symbols[0] / 4))
     final_state = backend.execute_circuit(
         c, initial_state=np.copy(initial_state), nshots=1
-    )[0]
+    )
 
     backend.set_seed(123)
-    c = models.Circuit(4)
+    c = models.Circuit(4, density_matrix=True)
     m = c.add(gates.M(1, collapse=True))
     target_state = backend.execute_circuit(
         c, initial_state=np.copy(initial_state), nshots=1
-    )[0]
+    ).state()
     if int(m.symbols[0].outcome()):
-        c = models.Circuit(4)
+        c = models.Circuit(4, density_matrix=True)
         c.add(gates.RY(0, theta=np.pi / 5))
         c.add(gates.RX(2, theta=np.pi / 4))
         target_state = backend.execute_circuit(c, initial_state=target_state)
     backend.assert_allclose(final_state, target_state)
 
 
+@pytest.mark.skip(reason="no way of currently testing this")
```

Comment:
For this and other tests that are skipped, we should decide whether:
* we keep as skipped but write a more explicit reason, for example: skipped because we no longer support repeated execution for density matrices (or something like that), or
* remove completely.

Either solution is fine with me.